programs/sponsored-cctp-src-periphery/src/utils/sponsored_cctp_quote.rs

```diff
@@ -117,171 +42,96 @@ impl<'a> SponsoredCCTPQuote<'a> {
+        Self::encode_pubkey(&mut encoded, &self.burn_token);
+        Self::encode_pubkey(&mut encoded, &self.destination_caller);
+        Self::encode_u64(&mut encoded, self.max_fee);
+        Self::encode_u32(&mut encoded, self.min_finality_threshold);
+
+        keccak::hash(&encoded).to_bytes()
     }
 
     /// `hash2` (EVM: second part) = keccak256(abi.encode(nonce, deadline, maxBpsToSponsor, maxUserSlippageBps,
     ///   finalRecipient, finalToken, executionMode, keccak256(actionData)))
     ///
-    /// We hash the static words (`Nonce..ExecutionMode`) directly from the head with appended `keccak(actionData)` as a
+    /// We hash the static words (`nonce..executionMode`) directly from the head with appended `keccak(actionData)` as a
     /// `bytes32` (static) value.
-    fn hash2(&self) -> Result<[u8; 32]> {
-        let start = SponsoredCCTPQuoteFields::Nonce.start();
-        let end = SponsoredCCTPQuoteFields::ExecutionMode.end();
-
-        // Safe: start and end are derived from SponsoredCCTPQuoteFields, so this should always be in-bounds.
-        Ok(keccak::hashv(&[&self.data[start..end], &self.action_data_hash()?]).to_bytes())
-    }
+    fn hash2(&self) -> [u8; 32] {
+        // Encode the following 7 static quote data fields + action_data hash.
+        let mut encoded = Vec::with_capacity(8 * 32);
 
-    /// `keccak256(actionData)` — hashes only the dynamic bytes content (not ABI-encoded), matching the Solidity side's
-    /// `keccak256(quote.actionData)`.
-    fn action_data_hash(&self) -> Result<[u8; 32]> {
-        // actionData bytes start immediately after its length word.
-        let start = SponsoredCCTPQuoteFields::ActionDataLength.end();
-        let length = self.get_action_data_len_checked()?;
+        Self::encode_bytes32(&mut encoded, &self.nonce);
+        Self::encode_u64(&mut encoded, self.deadline);
+        Self::encode_u64(&mut encoded, self.max_bps_to_sponsor);
+        Self::encode_u64(&mut encoded, self.max_user_slippage_bps);
+        Self::encode_pubkey(&mut encoded, &self.final_recipient);
+        Self::encode_pubkey(&mut encoded, &self.final_token);
+        Self::encode_u8(&mut encoded, self.execution_mode);
+        Self::encode_bytes32(&mut encoded, &keccak::hash(&self.action_data).to_bytes());
 
-        // Safe: get_action_data_len_checked() ensures that the actionData bytes are within bounds.
-        Ok(keccak::hash(&self.data[start..start + length]).to_bytes())
+        keccak::hash(&encoded).to_bytes()
     }
 
-    pub fn source_domain(&self) -> Result<u32> {
-        Self::decode_to_u32(self.get_field_word(SponsoredCCTPQuoteFields::SourceDomain))
-    }
-
-    pub fn destination_domain(&self) -> Result<u32> {
-        Self::decode_to_u32(self.get_field_word(SponsoredCCTPQuoteFields::DestinationDomain))
-    }
+    pub fn encode_hook_data(&self) -> Vec<u8> {
+        // ABI encoded hookData on EVM holds 7 static 32-byte words followed by the actionData offset that points to the
```

Comment:
Comment says “7 static words,” but code uses 8 * 32 (including executionMode)

is this correct?